crates/flux-infer/src/infer.rs

```diff
@@ -58,6 +59,10 @@ pub enum ConstrReason {
     Goto(BasicBlock),
     Overflow,
     Other,
+    SubtypeIn,
+    SubtypeReq,
+    SubtypeOut,
+    SubtypeEns,
```

Comment:
Maybe we can have a separate enum as a sort of *subreason*. Not sure though.

Also, in general, I was thinking that we can improve the errors at function calls because we have access to the span of the (actual) argument

Reply:
good idea -- let me spin this out into a separate issue (improving errors at calls).